### ISSUE TITLE
Avoid annoying "new version" window on startup

### ISSUE THREAD

**Issue by Kordonme** [ShareX/ShareX#527]
I would like to see a silent updater OR for ShareX to stop asking me on EVERY single boot. I'm gonna update when I feel like it. Or at least make me press that button and don't bug me anymore. Just handle it yourself - or stop asking me altogether ;)

Is it possible without disabling check for updates? Don't wanna miss out.

**Comment by Jaex** [comment]
Update operation already silent enough how can I make it more silent. Other apps even want you download it from url then do setup manually.

**Comment by Kordonme** [comment]
4 windows appear before the update is actually done. How is it not possible to simply:

1) check for update
2) download update
3) silent update

**Comment by Jaex** [comment]
You pressing "Yes" when it asks. It auto downloads, auto installs, auto opens it without even require you clicking anything.
So you tell me to not ask for it? Then countless people come here and open issues why it updates without asking...

**Comment by Kordonme** [comment]
You raise a valid point, but the default settings should be as currently, but a settings to make the auto updater would be appreciated. But I can see that's not happening anytime soon...

**Comment by McoreD** [comment]
I think ShareX can be updated the same way Chrome updates. Silently. Case:
ShareX is running. In background, ShareX has found that it has a newer
version. It does not prompt the user. When ShareX closes, it downloads and
run the installer. The problem with this approach is that, ShareX is hardly
closed by users because it is a tray app. It gets closed when OS is
rebooted or shutdown. This is not an ideal time to download and run the
installer. So alternatively, ShareX downloads the update to
%LocalAppData%\ShareX\Updates. When ShareX opens up next time, it can check
this location for updates and if there is an exe, it runs it and later
deletes this exe.

**Comment by jpmorris** [comment]
I think the problem is that you default the program to starting at login + have a window reminder.  Not many programs do this.  So unlike, say, microsoft word, adobe flash, acrobat, windows, chrome, firefox, skype.  It's not as if you're constantly hounded by a window at every boot to update.  This makes it far more noticeable.  I THINK this is the reason, but if I'm wrong I do know that ShareX is one of the most annoying programs when it comes to updating (it could also be the frequency?) But you should 1) silent install or 2) create a less intrusive update reminder.  Look how other programs are doing it. For silent install: Chrome, Windows.  They don't say "Oh well if I make a silent install then people will complain about the silent install."  Yes they will, and it's STILL a better solution.  You want these people to get the update, they don't really have much of a reason to complain about getting them, so you COULD just bite the bullet and make them take their medicine.  There's a reason many companies decided to go that way.  Option 2 is to do like Nvidia, Skype, acrobat, drivers of various sorts, Firefox, actually almost everything .  You create a little unobtrustive popup reminder or something part of the UI interface.  If it is a window  it should only come up when user EXPLICITLY opens the program (and should happen only after maybe every 2 months).  But the way you have ti setup I reboot my machine: HERE'S THE SHAREX WINDOW AND GREY BOX TO UPDATE. No time, close both, reboot: HERE'S THE SHAREX WINDOW AND GREY BOX TO UPDATE. Ugg, no.  Reboot:  HERE'S THE SHAREX WINDOW AND GREY BOX TO UPDATE. Ugg stop it!   

Again I might be wrong about my understanding somewhere here, but I do know that ShareX is one of the worst programs when it comes to update reminders.  I'm not 100% sure exactly what it is (timing? frequency? number of windows? not a popup? not part of the ui? required user interaction (unlike many other programs) color/size of the box? when it occurs? autostart? happens at each boot? happens at each program launch?), but I do know that it is.

**Comment by Jaex** [comment]
@jpmorris if you don't like ShareX feel free to not use it. I'm tired of people whining about it. Last 2.5 months whole day I worked to make this update and you can't even waste your few seconds to update it. Also don't make me laugh about telling how Chrome and Windows have good update systems. I'm not Google nor Microsoft, I'm working alone in this project and not earning anything other than donations. Can't handle everything myself. Also every single update system you offered using their own dedicated servers to handle updating. I don't even have hosting.

**Comment by McoreD** [comment]
Well said.

**Comment by Kordonme** [comment]
That response alone is enough to make me not want to use ShareX. It's an arrogant response. People here are trying to help by sharing their ideas and experiences with your product.

Seems like you are not made for the OS community. Good luck with your product. Finding an alternative or coding one myself.

**Comment by McoreD** [comment]
@Kordonme that response by @Jaex was clearly directed at @jpmorris. You cannot deny the fact that he was having an attitude in his first reply as well and not suited for an OS environment either. It is indeed frustrating after nearly 3 months of development, all one could say is just his problem pressing 'yes' to the update.

Being a project member, I myself have requested this. It will happen when it happens.

**Comment by Kordonme** [comment]
I can relate to that, it must be frustrating. But it could also be a signal that you're focusing in a direction, other than your users. 

I'm not complaining about your decision, because I didn't pay for the product. I'm happy with any another aspect of the application. But I'm done installing yet another app that bugs me on boot. 

I hope that one day, you are willing to implement this. You can make your product the most feature packed, most awesome and simple UI. But I'm still annoyed every time I open my computer, I won't remember all that :-)

**Comment by Jaex** [comment]
ShareX is all about users, why you think ShareX have countless settings? Because I added most of people's request. Last 8 years I'm trying my best to make people's life easier without getting paid and this is what I get. I don't know single other free open source product who added that much requests to software. Not to mention I reply to most of issues under 1 minute. And still you can tell me I don't care about users? How I can care more? Also I was already added better update system which not shows popup but couldn't complete it in time because was too busy with annotation features etc.

This is screenshot of it:

![](https://dl.dropboxusercontent.com/u/14076298/ShareX/2016/05/ShareX.vshost_EFCqCf6sHK.png)

But you guys don't deserve that. Feel free to make your own app, this is what I did too, I was not happy with other people's software and forced to make my own. Lets see how much you will care about your users and their requests!

**Comment by jpmorris** [comment]
@McoreD What made you think that I was having an 'attitude'?  I really was just giving constructive criticism

@Jaex So your'e saying you had a feature, agree with points, developed it, you have a screenshot to prove it, but people can't have it because you don't like people telling you what you already agree with? Wow. Just wow.  Are you a 12 year old child?  Do you have psychological Issues? Maybe an anger management problem or narcissistic personality disorder?  People are just giving you good ideas.  You don't have to even listen to them.  You could just say, 'good idea, but I don't have the time yet, here's a screenshot on what we'll have eventually'. That's what most RATIONAL people would do.  But you're not rational are you?  You see 'suggestions' and then you bitch and give so many irrelevant reasons (don't have update servers? I can give you 20 small projects that do a better job than you at updates). .You're just wrong. Face it. You're just wrong. And what do you do? You give all these reasons AND THEN you essentially admit that you were wrong by posting a screenshot and saying 'well I was going to do what you guys wanted but I'm too much of a overacting jerk so I'm just going to act like a child and take my toys and go home'.  Wow. This goes so much more beyond your bad UI design. This is something about you and your personality and behavior and it's going to effect your life Good luck with your bad UI and your personality disorder.

**Comment by Jaex** [comment]
I have screenshot because I was started to make it but didn't have time to complete it. Because I was busy making annotation stuffs. So reason why it is not in current build because I couldn't complete it yet. I already told you this in my previous post, not sure what you talking about.

**Comment by jpmorris** [comment]
I'm talking about:

"I worked to make this update and you can't even waste your few seconds to update it "
and this:
"Also every single update system you offered using their own dedicated servers to handle updating. I don't even have hosting."

Which are COMPLETELY IRRELEVANT since you're implementing the VERY THING that this entire issue was about.  Why don't you just say 'good idea, it's in the works'  And then you say this:

"But you guys don't deserve that. Feel free to make your own app, this is what I did too, I was not happy with other people's software and forced to make my own. Lets see how much you will care about your users and their requests!"

I ask again. Are you a 12 year old?

**Comment by Jaex** [comment]
What I'm implementing is not same thing. Chrome updating silently using Windows service in background without show update in main window. I'm not implementing completely silent update system which require to have my own hosting for proper implementation. This is why I told I need hosting for such good update system.

So in this implementation I'm making you are still needed to press "yes" in main window. And you prefer completely silent update system like Chrome.

**Comment by jpmorris** [comment]
Read the title of this issue.  It's not about silent updates. That was just a possible solution. The issue is about the annoying update window, not that you should implement silent updates.  Yes silent updates were brought up as one of the 5 suggestions of how to handle this annoying box.  So who cares if you use a silent update or not.  I certainly don't, I gave you 4 other ways of how to solve the problem.

But that doesn't concern me as much as your handling of this issue.  You do know that almost any other developer on github wouldn't have reacted the way you did right?  Go ahead an click on the user input of me, anyone else in this thread, or anywhere else on github.  The developers are rational people and when they get suggestions they don't agree with they just say, I'm sorry I don't agree I won't be implementing that.  And most of the time they would like it but they don't have time or any number of reasons. But you started acting like a child.  It didn't have to come to this.  People were just giving you reasonable suggestions.  Just stand back and look at them:  yes the box is very annoying  Yes there are other solutions.  People aren't just bitching at you because they want to piss you off, they really are just letting you know how to make the product better.  But you flipped out in a way that 99% of people on github would not.  Any reasonable person would have just said,  'working on it thanks'. That's it. But not you.

**Comment by Jaex** [comment]
Maybe if you followed your own advices I didn't have to respond like that.

**Comment by jpmorris** [comment]
And what was it about my post that made you think it was 'acting like a child', hmm?  Was it the 'I might be wrong's.  Or was it the many reasonable examples?  Or was it the time I took to make sure you knew all the different ways you could solve the problem? Or was it the 'I THINK' and uncertainty about what exactly the problem was.  Or was it the measured 'I'm not 100% sure'.  

Face the facts. You had no reason to overreact.   You can read 'shareX is one of the worst programs when it comes to updating'  without emotion.  It's true.  It is one of the worst when it comes to updating.  That doesn't mean it's not a good program, and doesn't mean you have to take it personally and act like a child.

**Comment by Jaex** [comment]
I don't like when same thing repeated 100 times. Every week someone comes here (or with email or IRC) and tells silent update system is required like in Chrome. Like I don't know that. And in your wall of texts you were keep repeating same things over and over again like I don't know how all other softwares doing updating. Also you were being rude in each of your messages and was acting like child yourself too. But of course you won't admit any of it. Therefore I was replying with similar manners.

"ShareX is one of the worst programs when it comes to updating" I'm not agree with that. I'm using alot of software too and other than few big software like Chrome, ShareX still have pretty decent updating system. Of course having system like Chrome will be best but I have reasons why I can't implement it.

**Comment by jpmorris** [comment]
No that's you ASSUMING that I think you don't know how other softwares do updates.  I assume you do know, but suggesting what you should change your UI and which one to implement it.  I was rude in my messages only AFTER you started acting like a child and flipping out like an irrational jerk.

So if you don't agree with that you just say what you said above.  You don't deal with the way you did. Of course you'd be wrong.  ShareX is one fo the worst programs when it comes to updating.  That's just a fact, which is why 'get the same things over and over again'

#552
#1432 
#1028 

But if you don't agree. That's fine you don't have to agree.  But it is one of the worst.

**Comment by district10** [comment]
Annoyed too. Go `Application Settings...` -- `Advanced`, and switch `Application/AutoCheckUpdate` to false.

**Comment by PatronBernard** [comment]
You cannot reasonably think that an application that launches on startup can prompt the user for updates _every single time._  ShareX is the _only_ app on my system that does it, because it's the only one that appears on each boot (among all of the PC's I use). 

If other apps did this as well, computers would become unusable. This is improper design. The easiest solution could be a "don't bother me for the next <X amount of time>" checkbox.  If silent updates aren't possible (which I think is weird, because non-silent updates have to be hosted somewhere as well, no?) 

It is simply not user friendly. It's the only thing that bothers me with ShareX, and I'm clearly not the only one.

**Comment by dannoe** [comment]
> every single time

It's a bug. #1796

**Comment by PatronBernard** [comment]
Oh, in that case: case closed. At least when the bug gets fixed.

**Comment by dannoe** [comment]
Yes it's fixed. Just use the way described in #1796 to update your ShareX.
